packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js

```diff
@@ -0,0 +1,157 @@
+import moment from 'moment';
+import { withStyles } from '@material-ui/core/styles';
+import DownArrow from '@material-ui/icons/ArrowDropDown';
+import MuiSwitch from '@material-ui/core/Switch';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { DateRangePicker } from '../../components/DateRangePicker';
+import { getDefaultDates, getLimits, GRANULARITY_CONFIG } from '../../utils/periodGranularities';
+import { Content, ContentText } from './Styles';
+import { TUPAIA_ORANGE } from '../../styles';
+import { setDisplayedMapOverlay } from '../../actions';
+
+const Switch = withStyles({
+  switchBase: {
+    '&$checked': {
+      color: TUPAIA_ORANGE,
+    },
+    '&$checked + $track': {
+      backgroundColor: TUPAIA_ORANGE,
+    },
+  },
+  checked: {},
+  track: {},
+})(MuiSwitch);
+
+const ToolsWrapper = styled.div`
+  display: flex;
+`;
+
+const IconWrapper = styled.div`
+  display: flex;
+  border-left: 1px solid rgba(255, 255, 255, 0.5);
+  padding: 8px 0 0 5px;
+`;
+
+const MeasureDatePicker = styled.div`
+  pointer-events: auto;
+  background: #203e5c;
+  padding: 16px 8px;
+  border-bottom-left-radius: ${({ expanded }) => (!expanded ? '5px' : '0')};
+  border-bottom-right-radius: ${({ expanded }) => (!expanded ? '5px' : '0')};
+`;
+
+export const TitleAndDatePickerComponent = ({
+  mapOverlay,
+  onUpdateMeasurePeriod,
+  isExpanded,
+  isMeasureSelected,
+  toggleMeasures,
+  isMeasureLoading,
+  displayedMapOverlays,
+  onSetDisplayedMapOverlay,
+  // TODO: PHX-103: show both date pickers
+  showDatePickerOnlyAfterSecondTitle,
+}) => {
+  const [isSwitchedOn, setIsSwitchedOn] = useState(true);
+  useEffect(() => {
+    setIsSwitchedOn(displayedMapOverlays.includes(mapOverlay?.mapOverlayId));
+  }, [JSON.stringify(displayedMapOverlays)]);
+  const switchOnChange = () => {
+    if (isSwitchedOn) {
+      onSetDisplayedMapOverlay(
+        displayedMapOverlays.filter(mapOverlayId => mapOverlayId !== mapOverlay?.mapOverlayId),
+      );
+    } else {
+      onSetDisplayedMapOverlay([...displayedMapOverlays, mapOverlay?.mapOverlayId]);
+    }
+    setIsSwitchedOn(!isSwitchedOn);
+  };
+  const { periodGranularity, isTimePeriodEditable = true, name } = mapOverlay;
+  const defaultDates = getDefaultDates(mapOverlay);
+  const datePickerLimits = getLimits(periodGranularity, mapOverlay.datePickerLimits);
+  const showDatePicker = !!(isTimePeriodEditable && periodGranularity);
+  const updateMeasurePeriod = (startDate, endDate) => {
+    const period = GRANULARITY_CONFIG[periodGranularity].momentUnit;
+    onUpdateMeasurePeriod(moment(startDate).startOf(period), moment(endDate).endOf(period));
+  };
+  // Map overlays always have initial dates, so DateRangePicker always has dates on initialisation,
+  // and uses those rather than calculating it's own defaults
+  const startDate = mapOverlay?.startDate || defaultDates.startDate;
+  const endDate = mapOverlay?.endDate || defaultDates.endDate;
+  return (
```

Comment:
Empty lines make the code breathe, similar to paragraphs in text:

```suggestion

  return (
```

I suggest restructuring this component using the following structure:
1. All non-function variable declarations go at the top (`useState`/other hooks go first, then other variable destructuring/assignments). Use a blank line at the end of this block. Can use more blank lines inside the block to group related operations if desired (may not be required here)
2. We then place function declarations. Most of the times we separate each of them with blank lines. `useEffect` should go either first or last here (not in between), to make it stand out

We don't define the above in a specific style guide, but you'll see them used as common practices in our functional components